coupler/coupler_types.F90

```diff
@@ -412,6 +562,9 @@ subroutine coupler_type_copy_1d_2d(var_in, var_out, is, ie, js, je,&
       call mpp_error(FATAL, trim(error_header) // ' Number of output fields exceeds zero')
     endif
 
+    if(associated(var_in%bc) .eq. associated(var_in%bc_r4)) &
```

Comment:
could there be a chance both are unassociated?

Reply:
yeah definitely, thats why its comparing the logical return values. If they are both associated or both not associated it will error out, this way it'll ensure that the r4/r8 field types used are mutually exclusive.

Reply:
I put in a fix, the check for allocation needed to be after the checks for the optional `as_needed` argument. I also made updates so that it'll only check for one kind allocated if bc%set is true since it indicates initialization.